runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java

```diff
@@ -0,0 +1,359 @@
+
+	@Inject
+	private WorldHopperInputListener inputListener;
+
+	private ScheduledFuture<?> lookupFuture;
+	private WorldResult worldResult;
+
+	private int worldToHopTo = -1;
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		keyManager.registerKeyListener(inputListener);
+
+		lookupFuture = executorService.scheduleAtFixedRate(() ->
+		{
+			try
+			{
+				WorldResult worldResult = new WorldClient().lookupWorlds();
+				worldResult.getWorlds().sort(Comparator.comparingInt(World::getId));
+				this.worldResult = worldResult;
+			}
+			catch (IOException ex)
+			{
+				log.warn(null, ex);
+			}
+		}, 0, 30, TimeUnit.MINUTES);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		keyManager.unregisterKeyListener(inputListener);
+
+		lookupFuture.cancel(true);
+	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		int groupId = WidgetInfo.TO_GROUP(event.getActionParam1());
+		String option = event.getOption();
+
+		if (groupId == WidgetInfo.FRIENDS_LIST.getGroupId() || groupId == WidgetInfo.CLAN_CHAT.getGroupId() ||
+			groupId == WidgetInfo.CHATBOX.getGroupId() && !KICK_OPTION.equals(option) || //prevent from adding for Kick option (interferes with the raiding party one)
+			groupId == WidgetInfo.RAIDING_PARTY.getGroupId() || groupId == WidgetInfo.PRIVATE_CHAT_MESSAGE.getGroupId())
+		{
+			boolean after;
+
+			if (AFTER_OPTIONS.contains(option))
+			{
+				after = true;
+			}
+			else if (BEFORE_OPTIONS.contains(option))
+			{
+				after = false;
+			}
+			else
+			{
+				return;
+			}
+
+			// Don't add entry if user is offline
+			ChatPlayer player = getChatPlayerFromName(event.getTarget());
+
+			if (player == null || player.getWorld() == 0 || player.getWorld() == client.getWorld())
+			{
+				return;
+			}
+
+			final MenuEntry hopTo = new MenuEntry();
+			hopTo.setOption(HOP_TO);
+			hopTo.setTarget(event.getTarget());
+			hopTo.setType(MenuAction.RUNELITE.getId());
+			hopTo.setParam0(event.getActionParam0());
+			hopTo.setParam1(event.getActionParam1());
+
+			insertMenuEntry(hopTo, client.getMenuEntries(), after);
+		}
+	}
+
+	private void insertMenuEntry(MenuEntry newEntry, MenuEntry[] entries, boolean after)
+	{
+		MenuEntry[] newMenu = ObjectArrays.concat(entries, newEntry);
+
+		if (after)
+		{
+			int menuEntryCount = newMenu.length;
+			ArrayUtils.swap(newMenu, menuEntryCount - 1, menuEntryCount - 2);
+		}
+
+		client.setMenuEntries(newMenu);
+	}
+
+	@Subscribe
+	public void onHopToMenuClicked(PlayerMenuOptionClicked event)
+	{
+		if (!event.getMenuOption().equals(HOP_TO))
+		{
+			return;
+		}
+
+		ChatPlayer player = getChatPlayerFromName(event.getMenuTarget());
+
+		if (player != null)
+		{
+			hop(player.getWorld());
+		}
+	}
+
+	void hop(boolean previous)
+	{
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			return;
+		}
+
+		int totalLevel = client.getTotalLevel();
+
+		World currentWorld = worldResult.findWorld(client.getWorld());
```

Comment:
This should account for maybe worldResult being null if it fails to load on plugin start.

Reply:
Added a nullcheck above